resources/keywords/login.resource

```diff
@@ -38,21 +62,30 @@ Login To Openshift
     Element Should Be Visible    ${txtBxUserName}
     Element Should Be Visible    ${txtBxPassword}
     Element Should Be Visible    ${btnLogin}
-    Input Text    ${txtBxUserName}    ${OCP_ADMIN_USER.USERNAME}
-    Input Password    ${txtBxPassword}    ${OCP_ADMIN_USER.PASSWORD}
+    Run Keyword If    ${DBaaSPolicyEnabled}    Run Keywords
+    ...    Input Text    ${txtBxUserName}    ${OCP_LDAP_USER.USERNAME}
+    ...    AND    Input Password    ${txtBxPassword}    ${OCP_LDAP_USER.PASSWORD}
+    ...  ELSE    Run Keywords
+    ...    Input Text    ${txtBxUserName}    ${OCP_ADMIN_USER.USERNAME}
+    ...    AND    Input Password    ${txtBxPassword}    ${OCP_ADMIN_USER.PASSWORD}
     Click Element    ${btnLogin}
     Page Should Not Contain Element    ${lblLoginError}    message=Login Failed
     Wait Until Keyword Succeeds    6x    5s    Wait Until Page Contains Element    ${btnPerspective}
+    User Closes Welcome Message
     User Selects Administrator View
 
 User Selects ${perspective} View
-    Wait Until Page Contains Element    ${btnPerspective}   timeout=15s
-    Click Element    ${btnPerspective}
-    ${elemPerspectiveOptUpd}=    Replace String    ${elemPerspectiveOpt}    <<option>>    ${perspective}
-    Wait Until Keyword Succeeds    6x    2s    Wait Until Page Contains Element  ${elemPerspectiveOptUpd}
-    Click Element    ${elemPerspectiveOptUpd}
+    ${elemCurPerspectiveUpd} =    Replace String    ${elemCurrPerspective}    <<perspective>>    ${perspective}
+    ${status} =    Run Keyword And Return Status    Element Should Be Visible    ${elemCurPerspectiveUpd}
+    IF    not ${status}
+        Wait Until Page Contains Element    ${btnPerspective}   timeout=15s
+        Click Element    ${btnPerspective}
+        ${elemPerspectiveOptUpd}=    Replace String    ${elemPerspectiveOpt}    <<option>>    ${perspective}
```

Comment:
Same as above

Reply:
I have shortened the variable to `elemCurPrspctvUpd`. Couldn't shorten it further without losing self description